src/OrchardCore/OrchardCore.DisplayManagement/BaseDisplayManager.cs

```diff
@@ -48,24 +49,56 @@ private static PlacementInfo FindPlacementImpl(ShapeTable shapeTable, string sha
                 shapeType = shapeType.Substring(0, delimiterIndex);
             }
 
-            if (shapeTable.Descriptors.TryGetValue(shapeType, out var descriptor))
-            {
-                var placementContext = new ShapePlacementContext(
-                    shapeType,
-                    displayType,
-                    differentiator,
-                    context.Shape
-                );
+            var placementContext = new ShapePlacementContext(
+                shapeType,
+                displayType,
+                differentiator,
+                context.Shape
+            );
 
-                var placement = descriptor.Placement(placementContext);
-                if (placement != null)
```

Comment:
Maybe some fast paths, here or before by the caller in `BindPlacementAsync()`

E.g.

            if (placementResolvers.Count == 1)
            {
                return placementResolvers[0].ResolvePlacement(placementContext);
            }